packages/cardpay-subgraph/src/mappings/register-reward-program-handler.ts

```diff
@@ -0,0 +1,33 @@
+import { RewardProgramRegistrationFee } from '../../generated/RewardProgram/RegisterRewardProgramHandler';
+import { log } from '@graphprotocol/graph-ts';
+
+import { RewardProgramRegistrationPayment } from '../../generated/schema';
+import { toChecksumAddress, makePrepaidCardPayment, makeToken, makeTransaction } from '../utils';
+
+export function handleRewardProgramRegistrationFee(event: RewardProgramRegistrationFee): void {
+  makeTransaction(event);
+  let prepaidCard = toChecksumAddress(event.params.prepaidCard);
+  let admin = toChecksumAddress(event.params.admin);
+  let rewardProgramID = toChecksumAddress(event.params.rewardProgramID);
```

Comment:
is a reward program ID actually an address? or is it a bytes32 field? only use `toChecksumAddress` for fields that are address types in solidity

Reply:
reward program ID is an address -- not bytes32. Are there concerns here? I use this simply as an id to represent reward program (so I generate it using `toChecksumAddress(randomHex(20))`. Any changes here will require a change quite a few things.

Reply:
It’s a little strange, but if it’s an address type in solidity then it’s ok to use the checksum address.